### ISSUE TITLE
validation requires electrode ElementIdentifiers to be int, but no such type in Matlab

### ISSUE THREAD

**Issue by emilyasterjones** [NeurodataWithoutBorders/matnwb#341]
This bug appears when validating `general_extracellular_ephys_electrodes`. Following the [MatNWB tutorial](https://neurodatawithoutborders.github.io/matnwb/tutorials/html/ecephys.html), which builds an electrodes table but doesn't manually specify the ids, you get:
```
ElementIdentifiers (general/extracellular_ephys/electrodes/id): incorrect type - expected 'int', got 'int8',
 DynamicTable/id (general/extracellular_ephys/electrodes/id): incorrect type - expected 'int', got 'int8']
```
Matlab only has int types where the number of bytes is specified (int8, int16, etc) so it's not possible for an int to be stored here.

**Comment by bendichter** [comment]
@emilyasterjones `int` is usually just shorthand for `int32`. The schema defines the minimum precision of a data type, so we need the data to be >= 32 bits. I think the problem here is that MatNWB is demoting integer types to `int8`. See #339

**Comment by ln-vidrio** [comment]
Does this still occur with the merging of #342?

**Comment by emilyasterjones** [comment]
Sorry, I don't have time to test this right now. If you use the code in the [tutorial](https://neurodatawithoutborders.github.io/matnwb/tutorials/html/ecephys.html) it should replicate the issue.

**Comment by ln-vidrio** [comment]
This appears to be fixed as well though I could be validating this incorrectly using PyNWB 2.0.0. Can @bendichter or @cechava confirm that this works?

**Comment by cechava** [comment]
@ln-vidrio this issue is fixed. The file now passes validation.

**Comment by cechava** [comment]
fixed with #342
